### PR TITLE
Unify the signature error when verifying web3 identities #2033

### LINKED ISSUE #2033: Unify the signature error when verifying web3 identities
### Context

Follow-up of #1989 

We have a unified interface to verify the LitentryMultiSignature, the error could have been unified too. We only keep it split for backwards compatibility:

https://github.com/litentry/litentry-parachain/blob/8553a33efaeedba48d2591b1ca6e032636830df8/tee-worker/app-libs/stf/src/helpers.rs#L162-L163

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by songzhouran**
### Context
fix #2033

### CODE REVIEW

**Review by BillyWooo — APPROVED** [review]
LGTM

**Comment by kziemianek** [comment]
Looks like a breaking change to me, isn't it ?

File: tee-worker/app-libs/stf/src/helpers.rs [hunk 1]
```diff
@@ -163,14 +163,7 @@ pub fn verify_web3_identity(
 	//       will merge it to `VerifyWeb3SignatureFailed` after the campaign
```
Comment (kziemianek) [inline tee-worker/app-libs/stf/src/helpers.rs]: I think you can remove this comment ?

**Review by kziemianek — CHANGES_REQUESTED** [review]

File: primitives/core/src/error.rs [hunk 2]
```diff
@@ -46,8 +46,7 @@ pub enum ErrorDetail {
 	WrongWeb2Handle,
 	UnexpectedMessage,
 	WrongSignatureType,
-	VerifySubstrateSignatureFailed,
-	VerifyEvmSignatureFailed,
+	VerifyWeb3SignatureFailed,
```
Comment (Kailai-Wang) [inline primitives/core/src/error.rs:49]: You need to regenerate the metadata for sure

File: tee-worker/app-libs/stf/src/helpers.rs [hunk 3]
```diff
@@ -163,14 +163,7 @@ pub fn verify_web3_identity(
...
 	if !data.signature().verify(raw_msg, identity) {
-		match data {
-			Web3ValidationData::Substrate(_) =>
-				return Err(StfError::LinkIdentityFailed(
-					ErrorDetail::VerifySubstrateSignatureFailed,
-				)),
-			Web3ValidationData::Evm(_) =>
-				return Err(StfError::LinkIdentityFailed(ErrorDetail::VerifyEvmSignatureFailed)),
-		}
+		return Err(StfError::LinkIdentityFailed(ErrorDetail::VerifyWeb3SignatureFailed))
```
Comment (Kailai-Wang) [inline tee-worker/app-libs/stf/src/helpers.rs]: I prefer using `ensure!`

**Review by Kailai-Wang — CHANGES_REQUESTED** [review]
You'd need to regenerate the metadata and types:
in the `tee-worker/ts-tests` run: `corepack yarn update-build`

**Review by kziemianek — APPROVED** [review]

**Review by Kailai-Wang — APPROVED** [review]

**Pull request merged**